### ISSUE TITLE
Help: Understadning Image  + Text Feature Fusion

### ISSUE THREAD

**Issue by mineshmathew** [facebookresearch/pythia#186]
## ❓ Questions and Help
Hey,

This is regarding the way logits are computed for a Pythia/LoRRa style VQA model. 
Once we compute a joint embedding as in https://github.com/facebookresearch/pythia/blob/37b2dc5d745fd752ba0c574e246856f03e49fb6d/pythia/models/lorra.py#L55,  a joint embedding of question  + image features  + ocr tokens is obtained and is of size N x 10000 for LoRRA.

Now in the `LogitClassifier` I was wondering why is there a need to have two separate channels ( `text_val` and  `image_val` to compute the logits and finally sum the logit values. Why isnt the `joint_embedding ` passed through a single non linear layer + linear layer sequence to compute the logits?

I am referring to the forward() of this module here:
 https://github.com/facebookresearch/pythia/blob/37b2dc5d745fd752ba0c574e246856f03e49fb6d/pythia/modules/layers.py#L134

**Comment by apsdehal** [comment]
It is there due to historical reasons. At some point, I believe original Pythia authors tried to use pretrained linear layers to encode information specific to text in `f_o_text` and specific to image in `f_o_image`. We didn't end up using those in final model though. Also, empirically, this worked better in past then just a simple MLP.
